addons/survey/models/survey_survey.py

```diff
@@ -181,6 +185,42 @@ def _read_group_stage_ids(self, stages, domain, order):
         stage_ids = stages._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return stages.browse(stage_ids)
 
+    @api.model
+    def create(self, vals):
+        if 'certification_badge_id' in vals:
```

Comment:
I would suggest a small reorganisation
1/ call super
2/ if badge -> create definition, challenge, then challenge line (and do this in a tool method)